### PR TITLE
fix vendor/k8s.io/metrics/pkg/client/custom_metrics staticcheck

### PR DESCRIPTION
**Pull request by habibrosyad**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Fixes `vendor/k8s.io/metrics/pkg/client/custom_metrics` staticcheck errors:

```sh
vendor/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go:49:4: ineffective break statement. Did you mean to break out of the outer loop? (SA4011)
vendor/k8s.io/metrics/pkg/client/custom_metrics/versioned_client.go:38:2: var codecs is unused (U1000)
```

**Which issue(s) this PR fixes**:

Part of #92402

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func PeriodicallyInvalidate(cache AvailableAPIsGetter, interval time.Duration, s
 		case <-ticker.C:
 			cache.Invalidate()
 		case <-stopCh:
-			break
+			return
```
Comment (lavalamp) [inline staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go:49]: This changes behavior, is there no affected test? :/ (to be clear, the new behavior of actually exiting looks better)

cc @logicalhan

Comment (logicalhan) [reply staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go:49]: lgtm but yeah maybe we should add a test here.

**Review by logicalhan — COMMENTED** [review]
/cc @serathius

**Comment by habibrosyad** [comment]
/cc @liggitt 
Could this also be in v1.20?

**Comment by liggitt** [comment]
> Could this also be in v1.20?

sure, though it looks like a test was requested in https://github.com/kubernetes/kubernetes/pull/94081#discussion_r479386384 ?

**Comment by habibrosyad** [comment]
> sure, though it looks like a test was requested in [#94081 (comment)](https://github.com/kubernetes/kubernetes/pull/94081#discussion_r479386384) ?

Hmm, that's what I'm wondering because of the lgtm. Sure, I could add a test for this.

File: staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+func (c *fakeDiscovery) ServerGroups() (*metav1.APIGroupList, error) {
+	if c.groupList == nil {
+		return nil, errors.New("doesn't exist")
+	}
+	return c.groupList, nil
+}
+
+func TestPeriodicallyInvalidate(t *testing.T) {
```
Comment (habibrosyad) [inline staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client_test.go:43]: Added a test here for `PeriodicallyInvalidate`. Could you review again @logicalhan @lavalamp?

**Pull request closed without merging**